src/main.h

```diff
@@ -100,6 +100,8 @@ static const unsigned int BLOCK_DOWNLOAD_WINDOW = 1024;
 static const unsigned int DATABASE_WRITE_INTERVAL = 60 * 60;
 /** Time to wait (in seconds) between flushing chainstate to disk. */
 static const unsigned int DATABASE_FLUSH_INTERVAL = 24 * 60 * 60;
+/** Time to wait (in seconds) between writing wallet witness data to disk. */
+static const unsigned int WITNESS_WRITE_INTERVAL = 10 * 60;
```

Comment:
It's okay to flush witness data more often than once per hour, and the `DATABASE_WRITE_INTERVAL` should really have multiple meanings, so give the witness flush its own constant (10 minutes).

Reply:
Is it necessary to write the witness data and the chain index data at separate intervals, though?

The performance argument for batched database writes is just that there's an overhead to doing any write at all. If we're doing a write, then we might as well flush the block index state as well.

(This argument doesn't apply to DATABASE_FLUSH_INTERVAL because there's currently a conflation between flushing that state to disk vs the cache discard policy. I'll explain this better in #4701.)

Reply:
The wallet database and the index database are independent, so I don't see the advantage of flushing both at the same time.